addons/survey/static/src/js/survey_session_manage.js

```diff
@@ -272,14 +276,17 @@ publicWidget.registry.SurveySessionManage = publicWidget.Widget.extend({
     _getPreviousScreen: function () {
         if (this.currentScreen === 'userInputs' && this.isScoredQuestion) {
             return 'question';
-        } else if (this.currentScreen === 'results' ||
-                  (this.currentScreen === 'leaderboard' && !this.isScoredQuestion)) {
+        } else if ((this.currentScreen === 'results' && this.isScoredQuestion) ||
+                  (this.currentScreen === 'leaderboard' && !this.isScoredQuestion) ||
+                  (this.currentScreen === 'leaderboardFinal' && this.isScoredQuestion)) {
             return 'userInputs';
-        } else if (this.currentScreen === 'leaderboard' && this.isScoredQuestion) {
+        } else if (this.currentScreen === "question" ||
```

Comment:
double quotes only for "question" ? unacceptable! :triangular_flag_on_post:

Reply:
:scream: